portal-impl/src/com/liferay/portlet/expando/service/impl/ExpandoValueLocalServiceImpl.java

```diff
@@ -463,8 +462,16 @@ public ExpandoValue addValue(
 			String columnName, long classPK, Object data)
 		throws PortalException {
 
-		ExpandoColumn column = _expandoColumnLocalService.getColumn(
-			companyId, className, tableName, columnName);
+		ExpandoColumn column = null;
```

Comment:
@shuyangzhou Although Lily inlined the original logic here, I wonder if `fetchByXXX` should be used. In this particular case, if table doesn't exist, it will result in a null `column`, causing NPE at line 476: `int type = column.getType()`; if the table exists, but the `column` doesn't, it will also cause NPE there.

I think both `fetchBy` should be replaced by `findBy` so `NoSuchXXXException` can be thrown with meaningful information.

Reply:
Yes, please do that, otherwise we could be facing some random NPEs.

Reply:
@slnn Could you please resend and apply the changes? Thanks!

Reply:
OK, got it

Reply:
Resend in https://github.com/liferay-core-infra/liferay-portal/pull/5532